test/pylib/cpp/common_cpp_conftest.py

```diff
@@ -105,13 +105,18 @@ def collect_items(file_path: PosixPath, parent: Collector, facade: CppTestFacade
     args = copy(DEFAULT_ARGS)
     custom_args_config = suite_config.get('custom_args', {})
     extra_scylla_cmdline_options = suite_config.get('extra_scylla_cmdline_options', [])
+    x_log2_compaction_groups = pytest_config.getoption('x_log2_compaction_groups', None)
+    all_can_run_compaction_groups_except = suite_config.get('all_can_run_compaction_groups_except', None)
     test_name = file_path.stem
+    allows_compaction_groups = all_can_run_compaction_groups_except and test_name not in all_can_run_compaction_groups_except
```

Comment:
IIUC, if the test suite doesn't have `all_can_run_compaction_groups_except` defined, we assign `None` to the `all_can_run_compaction_groups_except` variable here, and then `allows_compaction_groups` is set to `False`. It doesn't look right. I think we should do:
```python
allows_compaction_groups = not all_can_run_compaction_groups_except or (test_name not in all_can_run_compaction_groups_except)
```

Reply:
Yeah, I thought unset `all_can_run_compaction_groups_except` is supposed to mean what empty `all_can_run_compaction_groups_except` means. Now I see that unset `all_can_run_compaction_groups_except` is supposed to mean "no tests can run with compaction groups".

Quite confusing, but it was this way before your PR, so I won't complain.